### PR TITLE
Remote Sync | template | OKTA-377504

### PR DESCRIPTION
**Pull request by oleksandrkysil-okta**
Jira ticket:
- https://oktainc.atlassian.net/browse/OKTA-377504

Description:
- Reviewed template

Note:
- Currently without youtube video link, but the readme file can be reviewed.

### CODE REVIEW

File: workflows/remote_sync/readme.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Workflows Template: Remote Sync 
+
+
+## Overview
+
+Many CIAM customers have multiple user stores that need to be maintained until legacy systems can be decommissioned.
+When the identity information sourced in Okta changes these users and attributes need to be synchronized downstream. Okta Workflows provides an easy to implement, a fully customizable method to update a remote system with CRUD (create, update and delete) operations.
```
Comment (charleschan-okta) [inline workflows/remote_sync/readme.md:7]: wording: Okta Workflows provides an easy to implement and fully customizable method

File: workflows/remote_sync/readme.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    * Okta Workflow HTTP Connections securely enable;
+      * Basic Auth
+      * Custom Auth
+      * OAuth
+2. Select flow: "Event Update User of Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+3. Select flow: "Event Remove user From Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+4. Select flow: "Event Add user to Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
```
Comment (charleschan-okta) [inline workflows/remote_sync/readme.md:33]: Suggestion: For items 2, 3, and 4, can be more succinct:
For "Event Update User of Group", "Event Remove user From Group", and "Event Add user to Group", Make sure a connection is selected for the instances of the “Okta” cards. If it was not selected automatically, select the necessary connection manually.

File: workflows/remote_sync/readme.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+5. Select flow titled "[child]Create User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
+6. Select flow titled "[child]Update User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
+7. Select flow titled "[child]Delete User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
```
Comment (charleschan-okta) [inline workflows/remote_sync/readme.md:42]: Same suggestion here to be more succinct

File: workflows/remote_sync/workflow.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "remote_sync",
+  "title": "Remote Sync",
+  "description": "Many CIAM customers have multiple user stores that need to be maintained until legacy systems can be decommissioned. When the identity information sourced in Okta changes these attributes need to be synchronized downstream. Okta Workflows provides an easy to implement, fully customizable method to update a remote system with CRUD (create, update and delete) operations.",
```
Comment (charleschan-okta) [inline workflows/remote_sync/workflow.json:4]: wording: Okta Workflows provides an easy to implement and fully customizable method

**Review by charleschan-okta — CHANGES_REQUESTED** [review]

File: workflows/remote_sync/workflow.json (hunk 4 above)
Comment (markmoussa-okta) [reply workflows/remote_sync/workflow.json:4]: @oleksandrkysil-okta 
Please replace the description with this edited version:
Many CIAM customers have multiple user stores that need to be maintained until legacy systems are decommissioned. When the identity information sourced in Okta changes, these attributes need to be synchronized downstream. This template provides an easy-to-implement, fully customizable method to update a remote system with CRUD (create, update and delete) operations.

File: workflows/remote_sync/readme.md (hunk 1 above)
Comment (markmoussa-okta) [reply workflows/remote_sync/readme.md:7]: @oleksandrkysil-okta 
Please replace the summary in the Overview section with this edited version:
Many CIAM customers have multiple user stores that need to be maintained until legacy systems are decommissioned. When the identity information sourced in Okta changes, these attributes need to be synchronized downstream. This template provides an easy-to-implement, fully customizable method to update a remote system with CRUD (create, update and delete) operations.

File: workflows/remote_sync/readme.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+Many CIAM customers have multiple user stores that need to be maintained until legacy systems can be decommissioned.
+When the identity information sourced in Okta changes these users and attributes need to be synchronized downstream. Okta Workflows provides an easy to implement, a fully customizable method to update a remote system with CRUD (create, update and delete) operations.
+This integration uses Okta Group membership to identify those users that shall be synced. Adding a user to the group initiates the creation in the remote system. Removing a user from a group deletes the user from the remote system.
+In this sample, the user's name, address, and email are synced to the remote system. The flows can be modified to change attribute sync requirements. 
+
+The implementation uses a modular approach that splits the downstream CRUD operations into child flows to facilitate adaptation to complex environments.
+
+## Before you get Started / Prerequisites
+
+Before you get started, here are the things you’ll need:
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:15]: Remove "Before you get started, here are the things you’ll need:"

File: workflows/remote_sync/readme.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+*   Access to an Okta tenant with Okta Workflows enabled for your Org
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:17]: lower case "org"

File: workflows/remote_sync/readme.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+In this sample, the user's name, address, and email are synced to the remote system. The flows can be modified to change attribute sync requirements. 
+
+The implementation uses a modular approach that splits the downstream CRUD operations into child flows to facilitate adaptation to complex environments.
+
+## Before you get Started / Prerequisites
+
+Before you get started, here are the things you’ll need:
+
+*   Access to an Okta tenant with Okta Workflows enabled for your Org 
+*   The Remote System API calls in this sample are using a mock API. Create HTTP Connector for mock API Auth = none
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:18]: Replace the second sentence with:
"Create an HTTP connector with the Auth for the mock API as none."

File: workflows/remote_sync/readme.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+
+## Setup Steps
+
+1. Configure a HTTP Connector to communicate with a remote system
+    * Okta Workflow HTTP Connections securely enable;
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:24]: Enable secure HTTP connections:

File: workflows/remote_sync/readme.md (hunk 2 above)
Comment (markmoussa-okta) [reply workflows/remote_sync/readme.md:33]: Agree with Charles. Rewrite as follows:
For the following flows, make sure a connection is selected for the instances of the Okta connector cards:

- Event Update User of Group
- Event Remove User from Group
- Event Add User to Group

If the connection is not selected automatically, select the necessary connection manually.

Comment (markmoussa-okta) [reply workflows/remote_sync/readme.md:33]: Note that the flows should be renamed (capitalization-wise) as I suggest in my previous comment.

File: workflows/remote_sync/readme.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      * OAuth
+2. Select flow: "Event Update User of Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+3. Select flow: "Event Remove user From Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+4. Select flow: "Event Add user to Group".
+    * Make sure a connection is selected for the instances of the “Okta” cards.If it was not selected automatically, select the necessary connection manually.
+5. Select flow titled "[child]Create User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:36]: ...an appropriate HTTP connector...

File: workflows/remote_sync/readme.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+6. Select flow titled "[child]Update User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:39]: ...an appropriate HTTP connector...

File: workflows/remote_sync/readme.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+7. Select flow titled "[child]Delete User via API".
+    * Modify API URL and API Request JSON to match your remote system requirements
+   * Select appropriate HTTP Connector for "HTTP Raw Request" card
+8. Create Okta Group named "API Provisioning Group"
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:43]: Create an Okta group named...

File: workflows/remote_sync/readme.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+9. Add custom attribute to Okta User Profile name = 'customId' type=string
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:44]: Add a custom attribute to the Okta user profile...

File: workflows/remote_sync/readme.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+   * This attribute shall hold user Id from Remote System
```
Comment (markmoussa-okta) [inline workflows/remote_sync/readme.md:45]: This attribute will contain the user ID from...

File: workflows/remote_sync/readme.md (hunk 1 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: Did that, thanks

File: workflows/remote_sync/readme.md (hunk 2 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: Updated

File: workflows/remote_sync/readme.md (hunk 3 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: did that

File: workflows/remote_sync/workflow.json (hunk 4 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/workflow.json]: fixed

File: workflows/remote_sync/readme.md (hunk 5 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: removed

File: workflows/remote_sync/readme.md (hunk 6 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: added

File: workflows/remote_sync/readme.md (hunk 7 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: replaced

File: workflows/remote_sync/readme.md (hunk 8 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: changed

File: workflows/remote_sync/readme.md (hunk 9 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: changed

File: workflows/remote_sync/readme.md (hunk 10 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: changed

File: workflows/remote_sync/readme.md (hunk 11 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: fixed

File: workflows/remote_sync/readme.md (hunk 12 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: changed

File: workflows/remote_sync/readme.md (hunk 13 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: changed

File: workflows/remote_sync/workflow.json [hunk 14]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "remote_sync",
+  "title": "Remote Sync",
+  "description": "Many CIAM customers have multiple user stores that need to be maintained until legacy systems are decommissioned. When the identity information sourced in Okta changes, these attributes need to be synchronized downstream. This template provides an easy-to-implement, fully customizable method to update a remote system with CRUD (create, update and delete) operations.",
```
Comment (davidroossien-okta) [inline workflows/remote_sync/workflow.json:4]: (create, read, update and delete)

Comment (davidroossien-okta) [inline workflows/remote_sync/workflow.json:4]: replace CIAM with "customer identify and access management"

File: workflows/remote_sync/readme.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
+# Workflows Template: Remote Sync 
+
+
+## Overview
+
+Many CIAM customers have multiple user stores that need to be maintained until legacy systems are decommissioned. When the identity information sourced in Okta changes, these attributes need to be synchronized downstream. This template provides an easy-to-implement, fully customizable method to update a remote system with CRUD (create, update and delete) operations.
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:6]: replace CIAM with "customer identify and access management"

(create, read, update and delete)

File: workflows/remote_sync/readme.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+This integration uses Okta Group membership to identify those users that shall be synced. Adding a user to the group initiates the creation in the remote system. Removing a user from a group deletes the user from the remote system.
+In this sample, the user's name, address, and email are synced to the remote system. The flows can be modified to change attribute sync requirements. 
+
+The implementation uses a modular approach that splits the downstream CRUD operations into child flows to facilitate adaptation to complex environments.
+
+## Before you get Started / Prerequisites
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:12]: Remove "Before you get Started /"

File: workflows/remote_sync/readme.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+*   Access to an Okta tenant with Okta Workflows enabled for your org 
+*   The Remote System API calls in this sample are using a mock API. Create an HTTP connector with the Auth for the mock API as none.
+
+
+## Setup Steps
+
+1. Configure a HTTP Connector to communicate with a remote system
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:20]: should it be "Configure all the HTTP Connector cards"? 

which kind of card(s) are used in the flows? i.e. PUT, POST, DELETE? or CALL?

File: workflows/remote_sync/readme.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+6. Ensure all these flows are turned on:
+    *   Event Update User of Group
+    *   Event Remove user From Group
+    *   Event Add user to Group
+    *   [child]Update User Via API
+    *   [child]Delete User Via API
+    *   [child]Create User Via API
+
+
+## Testing this Flow
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:49]: "Testing these Flows"

File: workflows/remote_sync/readme.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
+The easiest way to test a flow is to add your test user to the Okta Group "API Provisioning Group"
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:51]: test these flows

File: workflows/remote_sync/readme.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+1. In the Okta Admin UI, navigate to the test users page.
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:53]: user's

File: workflows/remote_sync/readme.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+2. In the Groups tab of the Users page, add "API Provisioning Group".
+3. Inspect the remote system for the test user.
+4. In the Profile tab of the Users page, edit the value for "city" attribute.
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:56]: user's

File: workflows/remote_sync/readme.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+5. Inspect the remote system for change in test users' "city" value.
```
Comment (davidroossien-okta) [inline workflows/remote_sync/readme.md:57]: user's

**Review by davidroossien-okta — CHANGES_REQUESTED** [review]

File: workflows/remote_sync/readme.md (hunk 17 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md:20]: Raw Request

File: workflows/remote_sync/workflow.json (hunk 14 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/workflow.json]: Updated

Comment (oleksandrkysil-okta) [reply workflows/remote_sync/workflow.json]: replaced

File: workflows/remote_sync/readme.md (hunk 15 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: replaced

File: workflows/remote_sync/readme.md (hunk 16 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: removed

File: workflows/remote_sync/readme.md (hunk 18 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: updated

File: workflows/remote_sync/readme.md (hunk 19 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: updated

File: workflows/remote_sync/readme.md (hunk 20 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: updated

File: workflows/remote_sync/readme.md (hunk 21 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: fixed

File: workflows/remote_sync/readme.md (hunk 22 above)
Comment (oleksandrkysil-okta) [reply workflows/remote_sync/readme.md]: fixed

**Review by davidroossien-okta — APPROVED** [review]

**Pull request merged**